### ISSUE TITLE
Fatal Exception: java.lang.SecurityException Settings key: <data_roaming> is only readable to apps with targetSdkVersion lower than or equal to: 32

### ISSUE THREAD

**Issue by ka7erina** [braintree/braintree_android#605]
### General information
SDK/Library version: 4.16.0
Environment: Production
Android Version and Device: Pixel 4a Android 13, Pixel 6 Pro Android 13,  Pixel 6 Android 13, Pixel 5 Android 13
Braintree dependencies:
com.braintreepayments.api.braintree-core
com.braintreepayments.api.data-collector
com.braintreepayments.api.google-pay
com.braintreepayments.api.three-d-secure

### Issue description

We are facing this crash on some of our devices in production with the following exception:
```
Fatal Exception: java.lang.SecurityException
Settings key: <data_roaming> is only readable to apps with targetSdkVersion lower than or equal to: 32
android.provider.Settings$NameValueCache.getStringForUser (Settings.java:3081)

android.provider.Settings$Global.getString (Settings.java:15837)
com.cardinalcommerce.a.setHint.<init> (Unknown Source:170)
com.cardinalcommerce.a.setFreezesText.<init> (Unknown Source:23)
com.cardinalcommerce.a.setLineHeight.cca_continue (setLineHeight.java:53)
com.cardinalcommerce.cardinalmobilesdk.Cardinal.configure (Cardinal.java:5039)
com.braintreepayments.api.CardinalClient.configurationCardinal (CardinalClient.java:62)
com.braintreepayments.api.CardinalClient.initialize (CardinalClient.java:21)
com.braintreepayments.api.ThreeDSecureClient$1.onResult (ThreeDSecureClient.java:171)
com.braintreepayments.api.ConfigurationLoader$1.onResult (ConfigurationLoader.java:52)
com.braintreepayments.api.HttpClient$2.run (HttpClient.java:123)
```

**Comment by ka7erina** [comment]
@scannillo can you or someone from your team give more information?

**Comment by sshropshire** [comment]
Hi @ka7erina thanks for using the Braintree SDK for Android. We're working with our 3rd Party MPI Cardinal on this and they've notified us of an upcoming release to fix this issue in the coming weeks.

**Comment by jaxdesmarais** [comment]
Hey @ka7erina -

This has been fixed by our 3rd Party MPI Cardinal. The fix has been released in [version 4.18.0](https://github.com/braintree/braintree_android/releases/tag/4.18.0). Please let us know if you continue to run into this issue with their fix included after updating the SDK.
